metrics/api/v1/model_handlers.go

```diff
@@ -313,15 +314,22 @@ func (a *Api) podListMetrics(request *restful.Request, response *restful.Respons
 	}
 	ns := request.PathParameter("namespace-name")
 	keys := []string{}
-	for _, podName := range strings.Split(request.PathParameter("pod-list"), ",") {
-		keys = append(keys, core.PodKey(ns, podName))
-	}
-	metricName := getMetricName(request)
-	if metricName == "" {
-		response.WriteError(http.StatusBadRequest, fmt.Errorf("Metric not supported: %v", request.PathParameter("metric-name")))
-		return
+	metricName := request.PathParameter("metric-name")
+	convertedMetricName := metricName
+	for idx, podName := range strings.Split(request.PathParameter("pod-list"), ",") {
+		key := core.PodKey(ns, podName)
+		if idx == 0 {
+			validMetricName, valid := a.validateMetricName(metricName, key)
```

Comment:
Agreed - I'll update to validate for any known metric names & it will return empty for those that are not populated.